src/imitation/algorithms/adversarial/common.py

```diff
@@ -427,6 +430,45 @@ def _torchify_array(self, ndarray: Optional[np.ndarray]) -> Optional[th.Tensor]:
         if ndarray is not None:
             return th.as_tensor(ndarray, device=self.reward_train.device)
 
+    def _get_log_policy_act_prob(
+        self,
+        obs_th: th.Tensor,
+        acts_th: th.Tensor,
+    ) -> th.Tensor:
+        """Evaluates the given actions on the given observations.
+
+        Args:
+            obs_th: A batch of observations.
+            acts_th: A batch of actions.
+
+        Returns:
+            A batch of log policy action probabilities.
+        """
+        log_policy_act_prob, log_policy_act_prob_th = None, None
+        if isinstance(self.gen_algo.policy, policies.ActorCriticPolicy):
+            # policies.ActorCriticPolicy has a concrete implementation of
+            # evaluate_actions to generate log_policy_act_prob given obs and actions.
+            _, log_policy_act_prob_th, _ = self.gen_algo.policy.evaluate_actions(
+                obs_th,
+                acts_th,
+            )
+        elif isinstance(self.gen_algo.policy, sac_policies.SACPolicy):
+            gen_algo_actor = self.gen_algo.policy.actor
+            # generate log_policy_act_prob from SAC actor.
+            mean_actions, log_std, _ = gen_algo_actor.get_action_dist_params(obs_th)
+            distribution = gen_algo_actor.action_dist.proba_distribution(
+                mean_actions,
+                log_std,
+            )
+            log_policy_act_prob_th = distribution.log_prob(acts_th)
+            # getting log probability of actions from a continuous distribution
+            # might result in NaN values.
+            nan_log_prob_idx = th.where(th.isnan(log_policy_act_prob_th))
+            log_policy_act_prob_th[nan_log_prob_idx] = LOG_PROB_CUTOFF
```

Comment:
I think it's clearer if we just add an:
```python
else:
    return None
```
branch here, and then can assume that `log_policy_act_prob_th` is non-None (and remove the initialization line).

Reply:
Good suggestion